### PR TITLE
fix: implement checks on deleting an event

### LINKED ISSUE #5822: Implement checks on deleting an event
**Is your feature request related to a problem? Please describe.**

Deleting of an event should be prevent by an API if any order has been placed for the event, in order to avoid fraud. Please refer to https://github.com/fossasia/open-event-server/issues/5805#issuecomment-483576022

**Describe the solution you'd like**

Implement checks on deleting event. An event shouldn't be deleted if it has any order associated with it.

**Additional context**

Working on it.

### PR DESCRIPTION
**Pull request by shreyanshdwivedi**
Fixes #5822 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Deleting of an event should be prevent by an API if any order has been placed for the event, in order to avoid fraud.

#### Changes proposed in this pull request:
 - Implement checks on deleting event. An event shouldn't be deleted if it has any order associated with it.

### CODE REVIEW

**Comment by iamareebjamal** [comment]
A test would be amazing

**Comment by shreyanshdwivedi** [comment]
@iamareebjamal I tried to write test for this and I'm using `db.session.query(Event).filter_by(id=event.id).delete()` to delete the event object. But I think this command directly interacts with db and so my checks are not being called. Can you please help me out to pass a delete request so that my tests interact with my code?

**Comment by mrsaicharan1** [comment]
Use save_to_db from the db helpers to save it under a test_request_context.
Do all the operations under that test_request_context as it will be
interacting with the test db throughout the test.

**Comment by shreyanshdwivedi** [comment]
@mrsaicharan1 @iamareebjamal I tried everything here but was unable to fix the tests. The main problem which is coming is that on calling `delete()`, the event is getting deleted. Idk why this is happening. Can you please help me out?

**Comment by iamareebjamal** [comment]
OK, remove the tests. We'll merge the checks and add tests later

**Comment by iamareebjamal** [comment]
`db.session.query(Event).filter_by(id=event.id).delete()`

You don't have to use this. This will obviously delete the event from DB. You have added the checks at API level, so you have to call the DELETE API and test that it does not get deleted. Secondly, this delete will hard delete the event, however we have to just set is_active=False, that too will be tested on API level. The query won't fail magically when checks are at API level

**Comment by shreyanshdwivedi** [comment]
>OK, remove the tests. We'll merge the checks and add tests later

@iamareebjamal I've removed tests for now. Please give it a look

File: app/api/events.py [hunk 1]
```diff
@@ -468,9 +468,12 @@ def before_update_object(self, event, data, view_kwargs):
         :param view_kwargs:
         :return:
         """
-
         if has_access('is_admin') and data.get('deleted_at') != event.deleted_at:
-            event.deleted_at = data.get('deleted_at')
+            if len(event.orders) != 0:
+                raise ForbiddenException({'source': ''},
+                                     "Event associated with orders cannot be deleted")
```
Comment (uds5501) [inline app/api/events.py]: @shreyanshdwivedi Hound violation.

Comment (shreyanshdwivedi) [reply app/api/events.py]: @uds5501 fixed :sweat_smile:

**Comment by kushthedude** [comment]
@shreyanshdwivedi squash commits

**Comment by shreyanshdwivedi** [comment]
@kushthedude I guess commits can be squashed by maintainers while merging. However once they review it, I'll squash it if they need it for merging :)

**Comment by kushthedude** [comment]
Cool

File: app/api/events.py [hunk 2]
```diff
@@ -468,9 +468,11 @@ def before_update_object(self, event, data, view_kwargs):
         :param view_kwargs:
         :return:
         """
-
         if has_access('is_admin') and data.get('deleted_at') != event.deleted_at:
-            event.deleted_at = data.get('deleted_at')
+            if len(event.orders) != 0:
```
Comment (iamareebjamal) [inline app/api/events.py]: Check for only active orders please

Comment (shreyanshdwivedi) [reply app/api/events.py]: @iamareebjamal We concluded these possibilities of scams -

 - User can create an event and people buy it  and user do not show up on the venue and deletes the event from eventyay just after it finishes, as event will become past event.

If we change the condition and only restrict events with active orders to be deleted, how can we avoid above mentioned scams?

Comment (iamareebjamal) [reply app/api/events.py]: But he will not be able to delete event if it has active orders? In what cases an order can be deleted by a user?

Comment (shreyanshdwivedi) [reply app/api/events.py]: An order can never be deleted by a user. We are not giving them this feature as of now

**Comment by shreyanshdwivedi** [comment]
@mrsaicharan1 @uds5501 please review

**Pull request merged**